pkg/webhooks/kafkacluster_validator_test.go

```diff
@@ -690,3 +693,83 @@ func TestCheckExternalListenerStartingPort(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckExternalListenerIngressControllerTargetPort(t *testing.T) {
+	testCases := []struct {
+		testName         string
+		kafkaClusterSpec v1beta1.KafkaClusterSpec
+		expected         field.ErrorList
+	}{
+		{
+			testName: "valid config: ingressControllerTargetPort is not defined",
+			kafkaClusterSpec: v1beta1.KafkaClusterSpec{
+				ListenersConfig: v1beta1.ListenersConfig{
+					ExternalListeners: []v1beta1.ExternalListenerConfig{
+						{
+							CommonListenerSpec: v1beta1.CommonListenerSpec{Name: "test-external1"},
+						},
+						{
+							CommonListenerSpec: v1beta1.CommonListenerSpec{Name: "test-external2"},
+						},
+					},
+				},
+			},
+			expected: nil,
+		},
+		{
+			testName: "valid config: ingressControllerTargetPort is defined and is valid for IstioIngress controller",
+			kafkaClusterSpec: v1beta1.KafkaClusterSpec{
+				ListenersConfig: v1beta1.ListenersConfig{
+					ExternalListeners: []v1beta1.ExternalListenerConfig{
+						{
+							CommonListenerSpec:          v1beta1.CommonListenerSpec{Name: "test-external1"},
+							IngressControllerTargetPort: util.Int32Pointer(int32(v1beta1.MaxWellKnownPort + rand.Intn(65536-v1beta1.MaxWellKnownPort))),
+						},
+						{
+							CommonListenerSpec:          v1beta1.CommonListenerSpec{Name: "test-external2"},
+							IngressControllerTargetPort: util.Int32Pointer(int32(v1beta1.MaxWellKnownPort + rand.Intn(65536-v1beta1.MaxWellKnownPort))),
+						},
+					},
+				},
+			},
+			expected: nil,
+		},
```

Comment:
**Question**: I may be missing something here, but wouldn't this test create 2 different external listeners trying to listen on the same containerport (1024+65536-1024==65536)?
Shouldn't we guard against that in the webhook as well?

Reply:
This part is not needed anymore with the latest api tag